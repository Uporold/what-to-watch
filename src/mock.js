// eslint-disable-next-line import/prefer-default-export
export const fakeMoviesList = [
  `Batman`,
  `Lord Of The Rings`,
  `Star Wars`,
  `Deadpool`,
];
