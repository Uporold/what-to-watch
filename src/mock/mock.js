// eslint-disable-next-line import/prefer-default-export
export const movies = [
  {
    name: "Gangs of new york",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg",
    backgroundColor: "#A6B7AC",
    description:
      "In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.",
    rating: 8.8,
    scoresCount: 370881,
    director: "Martin Scorsese",
    starring: ["Leonardo DiCaprio", "Cameron Diaz", "Daniel Day-Lewis"],
    runtime: 167,
    genre: "Crime",
    released: 2002,
    id: 1,
    isFavorite: false,
    videoLink:
      "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
    previewVideoLink:
      "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
  },
  {
    name: "Moonrise Kingdom",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Moonrise_Kingdom.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Moonrise_Kingdom.jpg",
    backgroundColor: "#D8E3E5",
    description:
      "A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.",
    rating: 7.9,
    scoresCount: 291183,
    director: "Wes Anderson",
    starring: ["Jared Gilman", "Kara Hayward", "Bruce Willis"],
    runtime: 94,
    genre: "Adventure",
    released: 2012,
    id: 2,
    isFavorite: false,
    videoLink: "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
    previewVideoLink:
      "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
  },
  {
    name: "Pulp Fiction",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Pulp_Fiction.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/pulp-fiction.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Pulp_Fiction.jpg",
    backgroundColor: "#795433",
    description:
      "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
    rating: 1.5,
    scoresCount: 1635992,
    director: "Quentin Tarantino",
    starring: ["John Travolta", "Uma Thurman", "Samuel L. Jackson"],
    runtime: 153,
    genre: "Crime",
    released: 1994,
    id: 3,
    isFavorite: false,
    videoLink: "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
    previewVideoLink:
      "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
  },
  {
    name: "Bronson",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/bronson.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/bronson.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/bronson.jpg",
    backgroundColor: "#73B39A",
    description:
      "A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.",
    rating: 3.6,
    scoresCount: 109661,
    director: "Nicolas Winding Refn",
    starring: ["Tom Hardy", "Kelly Adams", "Luing Andrews"],
    runtime: 92,
    genre: "Action",
    released: 2008,
    id: 4,
    isFavorite: false,
    videoLink: "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
    previewVideoLink:
      "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
  },
  {
    name: "Midnight Special",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Midnight_Special.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/midnight-special.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Midnight_Special.jpg",
    backgroundColor: "#828585",
    description:
      "A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.",
    rating: 3.3,
    scoresCount: 67815,
    director: "Jeff Nichols",
    starring: ["Michael Shannon", "Joel Edgerton", "Kirsten Dunst "],
    runtime: 112,
    genre: "Action",
    released: 2016,
    id: 5,
    isFavorite: false,
    videoLink: "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
    previewVideoLink:
      "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
  },
];
