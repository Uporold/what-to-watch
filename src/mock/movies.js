// eslint-disable-next-line import/prefer-default-export
export const movies = [
  {
    name: "Gangs of new york",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg",
    backgroundColor: "#A6B7AC",
    description:
      "In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.",
    rating: 8.8,
    scoresCount: 370881,
    director: "Martin Scorsese",
    starring: ["Leonardo DiCaprio", "Cameron Diaz", "Daniel Day-Lewis"],
    runtime: 167,
    genre: "Crime",
    released: 2002,
    id: 1,
    isFavorite: false,
    videoLink:
      "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
    previewVideoLink:
      "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
    reviews: [
      {
        id: 1,
        user: { id: 10, name: "Max" },
        rating: 2.8,
        comment:
          "I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.",
        date: "2020-08-21T07:00:10.763Z",
      },
      {
        id: 2,
        user: { id: 12, name: "Isaac" },
        rating: 4.1,
        comment:
          "I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.",
        date: "2020-08-16T07:00:10.763Z",
      },
    ],
  },
  {
    name: "Moonrise Kingdom",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Moonrise_Kingdom.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Moonrise_Kingdom.jpg",
    backgroundColor: "#D8E3E5",
    description:
      "A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.",
    rating: 7.9,
    scoresCount: 291183,
    director: "Wes Anderson",
    starring: ["Jared Gilman", "Kara Hayward", "Bruce Willis"],
    runtime: 94,
    genre: "Adventure",
    released: 2012,
    id: 2,
    isFavorite: false,
    videoLink: "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
    previewVideoLink:
      "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
    reviews: [
      {
        id: 1,
        user: { id: 18, name: "Sophie" },
        rating: 3.4,
        comment:
          "It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.",
        date: "2020-08-26T07:00:10.763Z",
      },
      {
        id: 2,
        user: { id: 11, name: "Jack" },
        rating: 9.5,
        comment:
          "This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.",
        date: "2020-08-10T07:00:10.763Z",
      },
    ],
  },
  {
    name: "Pulp Fiction",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Pulp_Fiction.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/pulp-fiction.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Pulp_Fiction.jpg",
    backgroundColor: "#795433",
    description:
      "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
    rating: 1.5,
    scoresCount: 1635992,
    director: "Quentin Tarantino",
    starring: ["John Travolta", "Uma Thurman", "Samuel L. Jackson"],
    runtime: 153,
    genre: "Crime",
    released: 1994,
    id: 3,
    isFavorite: false,
    videoLink: "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
    previewVideoLink:
      "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
    reviews: [
      {
        id: 1,
        user: { id: 15, name: "Kendall" },
        rating: 8.1,
        comment:
          "This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.",
        date: "2020-08-04T07:00:10.763Z",
      },
      {
        id: 2,
        user: { id: 17, name: "Emely" },
        rating: 1.5,
        comment:
          "Unfortunately we don't have a reliable way to tell the true ratings of a movie.",
        date: "2020-08-23T07:00:10.763Z",
      },
      {
        id: 3,
        user: { id: 18, name: "Sophie" },
        rating: 5.8,
        comment:
          "Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.",
        date: "2020-07-29T07:00:10.764Z",
      },
    ],
  },
  {
    name: "Bronson",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/bronson.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/bronson.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/bronson.jpg",
    backgroundColor: "#73B39A",
    description:
      "A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.",
    rating: 3.6,
    scoresCount: 109661,
    director: "Nicolas Winding Refn",
    starring: ["Tom Hardy", "Kelly Adams", "Luing Andrews"],
    runtime: 92,
    genre: "Action",
    released: 2008,
    id: 4,
    isFavorite: false,
    videoLink: "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
    previewVideoLink:
      "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
    reviews: [
      {
        id: 1,
        user: { id: 13, name: "Zak" },
        rating: 3.7,
        comment:
          "Unfortunately we don't have a reliable way to tell the true ratings of a movie.",
        date: "2020-07-29T07:00:10.764Z",
      },
      {
        id: 2,
        user: { id: 18, name: "Sophie" },
        rating: 4.6,
        comment:
          "I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.",
        date: "2020-08-24T07:00:10.764Z",
      },
      {
        id: 3,
        user: { id: 14, name: "Corey" },
        rating: 7.4,
        comment:
          "This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.",
        date: "2020-08-07T07:00:10.764Z",
      },
    ],
  },
  {
    name: "Midnight Special",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Midnight_Special.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/midnight-special.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Midnight_Special.jpg",
    backgroundColor: "#828585",
    description:
      "A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.",
    rating: 3.3,
    scoresCount: 67815,
    director: "Jeff Nichols",
    starring: ["Michael Shannon", "Joel Edgerton", "Kirsten Dunst "],
    runtime: 112,
    genre: "Action",
    released: 2016,
    id: 5,
    isFavorite: false,
    videoLink: "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
    previewVideoLink:
      "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
    reviews: [],
  },
  {
    name: "Miidnight Special",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Midnight_Special.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/midnight-special.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Midnight_Special.jpg",
    backgroundColor: "#828585",
    description:
      "A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.",
    rating: 3.3,
    scoresCount: 67815,
    director: "Jeff Nichols",
    starring: ["Michael Shannon", "Joel Edgerton", "Kirsten Dunst "],
    runtime: 112,
    genre: "Action",
    released: 2016,
    id: 6,
    isFavorite: false,
    videoLink: "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
    previewVideoLink:
      "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
    reviews: [],
  },
  {
    name: "Miiidnight Special",
    posterImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Midnight_Special.jpg",
    previewImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/midnight-special.jpg",
    backgroundImage:
      "https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Midnight_Special.jpg",
    backgroundColor: "#828585",
    description:
      "A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.",
    rating: 3.3,
    scoresCount: 67815,
    director: "Jeff Nichols",
    starring: ["Michael Shannon", "Joel Edgerton", "Kirsten Dunst "],
    runtime: 112,
    genre: "Action",
    released: 2016,
    id: 7,
    isFavorite: false,
    videoLink: "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
    previewVideoLink:
      "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
    reviews: [],
  },
];
