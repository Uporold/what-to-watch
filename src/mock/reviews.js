export const reviews = [
  {
    id: 1,
    user: { id: 13, name: "Zak" },
    rating: 3.7,
    comment:
      "Unfortunately we don't have a reliable way to tell the true ratings of a movie.",
    date: "2020-07-29T07:00:10.764Z",
  },
  {
    id: 2,
    user: { id: 18, name: "Sophie" },
    rating: 4.6,
    comment:
      "I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.",
    date: "2020-08-24T07:00:10.764Z",
  },
  {
    id: 3,
    user: { id: 14, name: "Corey" },
    rating: 7.4,
    comment:
      "This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.",
    date: "2020-08-07T07:00:10.764Z",
  },
];
