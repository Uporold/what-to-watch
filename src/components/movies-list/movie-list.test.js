import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "./movies-list";
import { movies } from "../../mock/mock";

it(`Should Movies List render correctly`, () => {
  const tree = renderer
    .create(<MoviesList movies={movies} onSmallCardMovieClick={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
