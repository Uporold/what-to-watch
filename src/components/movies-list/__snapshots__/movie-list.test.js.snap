// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Movies List render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Gangs of new york"
        height="175"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Gangs of new york
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Moonrise Kingdom"
        height="175"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Moonrise Kingdom
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Pulp Fiction"
        height="175"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/pulp-fiction.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Pulp Fiction
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bronson"
        height="175"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/bronson.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bronson
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Midnight Special"
        height="175"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/midnight-special.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Midnight Special
      </a>
    </h3>
  </article>
</div>
`;
